### PR TITLE
feat: validation & upload warnings [experience-6090]

### PR DESCRIPTION
**Pull request by doug-s-nava**
This change adds readouts on the validation and upload pages for warnings sent back from the API when validating or uploading a file.

Also includes:
* addition of `field` and `tracking ids` columns to errors table
* some styling fixes
* some text updates
* some renaming and splitting out functionality in file handler messaging

### TODOs

- [ ] confirm behavior around overlap between errors and warnings in csv response. This can happen later
- [x] design / content confirmation? (we will follow up with any necessary design and content changes later)

## Test Steps:

1. start servers on this branch with `yarn run start:localdev` from `frontend-react` and `./gradlew run` from `prime-router`
2. visit http://localhost:3000/admin/settings and log in as necessary
4. activate the `user-upload` feature flag via the `admin -> feature flags` navigation
6. refresh
3. set up your user up as a sender that is set up to send hl7 files, ex. `hcintegrations`
8. click `Validate` in the nav bar under `Upload & Validate`
10.  upload an hl7 file that will return warnings (you can use the third file available from this ticket https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/cdcgov/prime-reportstream/5709)
7. _VERIFY_: page looks like the screenshot below

## Screenshots
Screen Shot 2022-07-20 at 3 24 43 PM

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes 6090

### CODE REVIEW

File: frontend-react/src/components/FileHandlers/FileHandler.tsx [hunk 1]
```diff
@@ -259,29 +262,27 @@ const FileHandler = ({
             if (response?.id) {
                 setReportId(response.id);
                 setSuccessTimestamp(response.timestamp);
-                event.currentTarget.reset();
+                event?.currentTarget?.reset && event.currentTarget.reset();
```
Comment (doug-s-nava) [inline frontend-react/src/components/FileHandlers/FileHandler.tsx:265]: this was throwing a silent error when the currentTarget did not have a reset method

File: frontend-react/src/components/FileHandlers/FileHandler.tsx [hunk 2]
```diff
@@ -259,29 +262,27 @@ const FileHandler = ({
...
         } catch (error) {
             // Noop.  Errors are collected below
+            console.error("Unexpected error in file handler", error);
         }
 
         // Process the error messages
+        // TODO: deprecate FileResponseErrors
         if (response?.errors && response.errors.length > 0) {
             // Add a string to properly display the indices if available.
-            const FileResponseErrors = response.errors.map((error: any) => {
```
Comment (doug-s-nava) [inline frontend-react/src/components/FileHandlers/FileHandler.tsx:277]: moved to ErrorRow component

File: frontend-react/src/components/FileHandlers/FileHandlerMessaging.tsx [hunk 3]
```diff
@@ -92,35 +92,27 @@ export const FileSuccessDisplay = ({
...
-    )
-        return (
-            truncateErrorMessage.substring(
-                0,
-                truncateErrorMessage.indexOf("\n")
-            ) + " ..."
-        );
-
-    return truncateErrorMessage;
+const truncateErrorMesssage = (errorMessage: string | undefined): string => {
```
Comment (doug-s-nava) [inline frontend-react/src/components/FileHandlers/FileHandlerMessaging.tsx:101]: just renaming things

Comment (sethdarragile6) [reply frontend-react/src/components/FileHandlers/FileHandlerMessaging.tsx:101]: oh _that's_ what you meant! yeah, fire me...

**Review by sethdarragile6 — APPROVED** [review]
LGTM!

**Pull request merged**